qiskit/transpiler/__init__.py

```diff
@@ -40,629 +41,787 @@
-   move the qubit states around until they are adjacent on the device gate map.  Each SWAP
-   gate is decomposed into three CNOT gates on the IBM Quantum devices, and represents an
-   expensive and noisy operation to perform.  Thus, finding the minimum number of SWAP gates
-   needed to map a circuit onto a given device, is an important step (if not the most important)
-   in the whole execution process.
+      new_circ_lv3 = transpile(ghz, backend=backend, optimization_level=3)
+      plot_circuit_layout(new_circ_lv3, backend)
 
-   However, as with many important things in life, finding the optimal SWAP mapping is hard.
-   In fact it is in a class of problems called NP-Hard, and is thus prohibitively expensive
-   to compute for all but the smallest quantum devices and input circuits.  To get around this,
-   by default Qiskit uses a stochastic heuristic algorithm called
-   :class:`Qiskit.transpiler.passes.StochasticSwap` to compute a good, but not necessarily minimal
-   SWAP count.  The use of a stochastic method means the circuits generated by
-   :func:`Qiskit.compiler.transpile` (or :func:`Qiskit.execute` that calls `transpile` internally)
-   are not guaranteed to be the same over repeated runs.  Indeed, running the same circuit
-   repeatedly will in general result in a distribution of circuit depths and gate counts at the
-   output.
 
-   In order to highlight this, we run a GHZ circuit 100 times, using a "bad" (disconnected)
-   `initial_layout`:
+It is completely possible to specify your own initial layout.  To do so we can
+pass a list of integers to :func:`qiskit.compiler.transpile` via the `initial_layout`
+keyword argument, where the index labels the virtual qubit in the circuit and the
+corresponding value is the label for the physical qubit to map onto:
 
-   .. jupyter-execute::
+.. jupyter-execute::
 
-      import matplotlib.pyplot as plt
-      from qiskit import QuantumCircuit, transpile
-      from qiskit.providers.fake_provider import FakeBoeblingen
-      backend = FakeBoeblingen()
+   # Virtual -> physical
+   #    0    ->    3
+   #    1    ->    4
+   #    2    ->    2
 
-      ghz = QuantumCircuit(5)
-      ghz.h(0)
-      ghz.cx(0,range(1,5))
-      ghz.draw(output='mpl')
+   my_ghz = transpile(ghz, backend, initial_layout=[3, 4, 2])
+   plot_circuit_layout(my_ghz, backend)
 
+.. _routing_stage:
 
-   .. jupyter-execute::
-
-      depths = []
-      for _ in range(100):
-         depths.append(transpile(ghz,
-                                 backend,
-                                 initial_layout=[7, 0, 4, 15, 19],
-                                ).depth())
+Routing Statge
+--------------
 
-      plt.figure(figsize=(8, 6))
-      plt.hist(depths, bins=list(range(14,36)), align='left', color='#AC557C')
-      plt.xlabel('Depth', fontsize=14)
-      plt.ylabel('Counts', fontsize=14);
+In order to implement a CNOT gate between qubits in a quantum circuit that are not directly
```

Comment:
Do we want to generalise this to "a 2-qubit gate"?  (Or perhaps even go further and refer to heterogeneous gate support again?)

Reply:
Fixed in: https://github.com/Qiskit/qiskit-terra/pull/9087/commits/a46e97de3261a7a394c0f63ff23f818f5936f49f